lib/elixir/lib/module/checker.ex

```diff
@@ -271,6 +299,65 @@ defmodule Module.Checker do
     ]
   end
 
+  defp format_warning({:unable_unify, left, right, expr, traces}) do
```

Comment:
My suggestion is for us to follow Erlang's approach here. i.e. each error is in the format `{module, value}` and you call `module.format_error(value)` to get the actual reason. This means we can keep all of the error handling in the `Types` module.